egui/src/widgets/slider.rs

```diff
@@ -199,6 +202,28 @@ impl<'a> Slider<'a> {
         self
     }
 
+    /// Sets the minimal change of the value.
+    /// If `clamp_to_range` is enabled, `step` must be less than a maximum between the start and
+    /// the end of the range; otherwise, the step would be left unchanged.
+    /// Without `clamp_to_range` enabled, `step` can be any value.
+    ///
+    /// Default: `None`.
+    pub fn step_by(mut self, step: f64) -> Self {
+        let max = self.range.end().abs().max(self.range.start().abs());
+        let step = match self.clamp_to_range {
```

Comment:
This means the order of calling `clamp_to_range` and `step_by` is important - better to just set the step here and resolve this in `set_value` instead imho.

Reply:
I've removed all clamping logic from here. Turned out, it's overengineering.
Also, I think that the implicit turn-off of the stepping is worse than being stuck at 0.0.